Add support for nrf54l15dk/nrf54l15/cpuflpr

masz-nordic opened the pull request:
Depends on:
https://github.com/zephyrproject-rtos/zephyr-lang-rust/pull/128
https://github.com/zephyrproject-rtos/zephyr-lang-rust/pull/108

cordawyn committed:
added support for more granular RISCV32/64 ISA extensions

masz-nordic committed:
dt-rust.yaml: add nRF54L flash controller

RRAM controller is used on the nRF54L series.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
samples: async-philosopers: add missing space

Space was missing before board name.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
samples: add support for nrf54l15dk/nrf54l15/cpuapp

Tested manually, all tests are passing.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
gitignore: skip `build` dirs

Ignore the build directory.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
cmake: add support for RV32E targets

Tier 3 no-std targets that require building core and alloc from source.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
{kconfig, cmake}: add options to build nightly and std from source

Some targets need to be built with nightly and `-Z build-std=core,alloc`

masz-nordic committed:
Kconfig: enable building for RISCV_ISA_RV32E

It is possible to build for that target with nightly and
standard library built from source.

masz-nordic committed:
samples: replace BOARD with BOARD_TARGET

BOARD_TARGET is more specific about which core runs the sample.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

masz-nordic committed:
samples: enable support for nrf54l15dk/nrf54l15/cpuflpr

Due to memory constraints, only samples built in release
configuration can be enabled.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

d3zd3z commented on samples/hello_world/src/lib.rs:
```diff
@@ -11,5 +11,5 @@ extern "C" fn rust_main() {
     }
 
-    info!("Hello world from Rust on {}", zephyr::kconfig::CONFIG_BOARD);
+    info!("Hello world from Rust on {}", zephyr::kconfig::CONFIG_BOARD_TARGET);
```
You'll need to either format these manual, or just use `cargo fmt` to reflow them. With the longer name, it breaks this across multiple lines.

d3zd3z reviewed (approved):
Looks good, and thank you for this contribution. This should help us get more targets that don't use build configurations that exactly match one tier 1 or 2 of Rust.

I think there is still a rustfmt issue with one of the longer names, but I'm not seeing it locally.

d3zd3z committed:
samples: hello_world: Fix formatting

The longer identifier for the board messages causes rustfmt to flow this
onto multiple lines.  Apply this change so that the check will pass
during CI.

Signed-off-by: David Brown <david.brown@linaro.org>
